File: python/45d_1527A_AndThenThereWereK.py
from math import log2, pow
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        k = int(pow(2, int(log2(n)))) - 1
        ans.append(k)
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/45d_1527A_AndThenThereWereK.cpp
#include<bits/stdc++.h>

int main(){
    int t, n, k;
    std::vector<int> ans;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        std::cin>>n;
        k = log2(n);
        k = pow(2, k);
        k--;
        ans.push_back(k);
    }
    for (auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}